src/RingGrids/grids/full_clenshaw.jl

```diff
@@ -0,0 +1,38 @@
+struct FullClenshawArray{T, N, ArrayType <: AbstractArray{T, N}} <: AbstractFullGridArray{T, N, ArrayType}
+    data::ArrayType                 # data array, ring by ring, north to south
+    nlat_half::Int                  # number of latitudes on one hemisphere
+    rings::Vector{UnitRange{Int}}   # TODO make same array type as data?
```

Comment:
Indices should always stay on the CPU as far as I know, so that should be fine (but might warrant a small test). (Same comment applies to all other grid types)

Reply:
Okay sounds good, worst case we can still change that later without many problems I believe as we mostly use `eachring(grid) = grid.rings` so there's room to do something else instead.